src/Microsoft.OData.Edm/Csdl/CsdlReader.Json.cs

```diff
@@ -0,0 +1,312 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="CsdlReader.Json.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETSTANDARD2_0
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.Json;
+using Microsoft.OData.Edm.Csdl.CsdlSemantics;
+using Microsoft.OData.Edm.Csdl.Parsing;
+using Microsoft.OData.Edm.Csdl.Parsing.Ast;
+using Microsoft.OData.Edm.Validation;
+
+namespace Microsoft.OData.Edm.Csdl
+{
+    /// <summary>
+    /// A delegate used to set the referenced CSDL-JSON reader.
+    /// </summary>
+    /// <param name="uri">The reference Uri.</param>
+    /// <param name="skip">out a boolean value that indicates whether to skip parsing the input uri.</param>
+    /// <returns>The corresponding <see cref="Utf8JsonReader"/>.</returns>
+    /// <remarks>
+    /// Utf8JsonReader is a ref struct, so we can't distiguish which Uri you want to parse, which Uri want to skip.
+    /// So, use an out boolean value to indicate whether to skip parsing the input reference Uri.
+    /// </remarks>
+    public delegate Utf8JsonReader CsdlJsonReaderFactory(Uri uri, out bool skip);
+
+    /// <summary>
+    /// Provides CSDL-JSON parsing services for EDM models.
+    /// </summary>
+    public partial class CsdlReader
+    {
+        /// <summary>
+        /// Tries parsing the given CSDL-JSON artifact for an IEdmModel.
+        /// </summary>
+        /// <param name="reader">The given JSON reader containing the CSDL artifact.</param>
+        /// <param name="model">The model generated by parsing</param>
+        /// <param name="errors">Errors reported while parsing.</param>
+        /// <returns>Success of the parse operation.</returns>
+        public static bool TryParse(ref Utf8JsonReader reader, out IEdmModel model, out IEnumerable<EdmError> errors)
+        {
+            return TryParse(ref reader, CsdlJsonReaderOptions.Default, out model, out errors);
+        }
+
+        /// <summary>
+        /// Tries parsing the given CSDL-JSON artifact for an IEdmModel.
+        /// </summary>
+        /// <param name="reader">The given JSON reader containing the CSDL artifact.</param>
+        /// <param name="ignoreUnexpecteJsonElements">Ignore the unexpected JSON element in schema.</param>
+        /// <param name="model">The model generated by parsing</param>
+        /// <param name="errors">Errors reported while parsing.</param>
+        /// <returns>Success of the parse operation.</returns>
+        public static bool TryParse(ref Utf8JsonReader reader, bool ignoreUnexpecteJsonElements, out IEdmModel model, out IEnumerable<EdmError> errors)

```

Comment:
Would it be an option to fold ignoreUnexpecteJsonElements into the CsdlJsonReaderOptions and therefore remove this overload.

Reply:
As discussed, i removed this overload, thanks.

---
In reply to: [425976026](https://github.com/OData/odata.net/pull/1775#discussion_r425976026) [](ancestors = 425976026)